crates/app-data/src/app_data.rs

```diff
@@ -18,7 +18,7 @@ pub struct ValidatedAppData {
     pub protocol: ProtocolAppData,
 }
 
-#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq)]
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
```

Comment:
These were added across a bunch of structures in order to "Serialize" the AppData structure as JSON for post.